game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTerritoryValueUtils.java

```diff
@@ -448,39 +459,26 @@ private static double findWaterValue(
    */
   protected static Collection<Territory> findNearbyEnemyCapitalsAndFactories(
       final Territory startTerritory, final Set<Territory> enemyCapitalsAndFactories) {
-    final GameMap map = startTerritory.getData().getMap();
-    // Use breadth first search to traverse territories, keeping track of which have already been
-    // visited and which territories from the target list have  been found.
     final var found = new HashSet<Territory>();
-    final var visited = new HashSet<Territory>(List.of(startTerritory));
-    final var territoriesToCheck = new ArrayDeque<Territory>(List.of(startTerritory));
-
-    // Since we process territories in order of distance, we can keep track of the last territory
-    // at the current distance that's in the territoriesToCheck queue. When we encounter it, we
-    // increment the distance and update lastTerritoryAtCurrentDistance.
-    int distance = 0;
-    Territory lastTerritoryAtCurrentDistance = startTerritory;
-    while (!territoriesToCheck.isEmpty()) {
-      final Territory node = territoriesToCheck.removeFirst();
-      for (final Territory neighbor : map.getNeighbors(node)) {
-        if (visited.add(neighbor)) {
-          territoriesToCheck.add(neighbor);
-          if (enemyCapitalsAndFactories.contains(neighbor)) {
-            found.add(neighbor);
+    final BreadthFirstSearch bfs =
```

Comment:
What do you think of using a var here in order to spell out the variable name instead? eg:
`var breadthFirstSearch =  new BreadthFirstSearch(startTerritory) {`

Reply:
Done.